### PR TITLE
Add PAT login prompt to launch script.

### PR DESCRIPTION
**Pull request by deeleeramone**
This PR:

- Adds a Hub login prompt (can be bypassed) to update `user_settings.json` with values from Hub.
- Prompt for persistence, if no the downloaded changes are removed on `KeyboardInterrupt`.
- Adds a check to determine if the port is actually free.
- When bad host/port values are in the ENV, adds an input prompt instead of raising.

### CODE REVIEW

File: openbb_platform_pro_backend/main.py [hunk 1]
```diff
@@ -11,6 +13,22 @@
     data_schema_to_columns_defs,
 )
 
+CURRENT_USER_SETTINGS = os.environ.get("HOME") + "/.openbb_platform/user_settings.json"
```
Comment (piiq) [inline openbb_platform_pro_backend/main.py]: to avoid potential errors on windows let's specify the path either with `Path` or as `os.path.join`

File: openbb_platform_pro_backend/main.py [hunk 2]
```diff
@@ -11,6 +13,22 @@
...
+USER_SETTINGS_COPY = CURRENT_USER_SETTINGS.replace("user_settings.json", "user_settings_copy.json")
```
Comment (piiq) [inline openbb_platform_pro_backend/main.py]: instead of `_copy` let's use `_backup`

File: openbb_platform_pro_backend/main.py [hunk 3]
```diff
@@ -99,25 +117,130 @@ async def get_widgets():
...
-    import os
     import uvicorn
 
+    if Path(CURRENT_USER_SETTINGS).exists():
+        with open(CURRENT_USER_SETTINGS, "r") as f:
+            current_settings = json.load(f)
+    else:
+        current_settings = {"credentials": {}, "preferences": {}, "defaults": {"commands": {}}}
+
+    pat = input(
```
Comment (piiq) [inline openbb_platform_pro_backend/main.py]: can you check if we can make the input be not fully visible by default? either not visible at all, like we usually see on linux when we enter the sudo password, or replacing each symbol with `*`

File: openbb_platform_pro_backend/main.py [hunk 4]
```diff
@@ -99,25 +117,130 @@ async def get_widgets():
...
+                    continue
+            new_settings["defaults"].update(hub_defaults)
+
+        # Write the new settings to the user_settings.json file
+        with open(CURRENT_USER_SETTINGS, "w") as f:
+            json.dump(new_settings, f, indent=4)
+
+        current_settings = new_settings
+
+    if current_settings["credentials"].get("OPENAI_API_KEY"):
```
Comment (piiq) [inline openbb_platform_pro_backend/main.py]: can you give a bit more context why we differentiate OpenAI from other keys? iiuc the platform api is not reliant on OpenAI at all, but I might be missing something

File: openbb_platform_pro_backend/main.py [hunk 5]
```diff
@@ -99,25 +117,130 @@ async def get_widgets():
...
+            "\n\nOPENBB_API_HOST is set incorrectly. It should be an IP address or hostname."
         )
+        host = input("Enter the host IP address or hostname: ")
+        if not host:
+            host = "127.0.0.1"
 
     port = os.getenv("OPENBB_API_PORT", 8000)
+
     try:
         port = int(port)
```
Comment (piiq) [inline openbb_platform_pro_backend/main.py:212]: when we check the port number we actually need to check that the port number is not bellow 1024 as those are usually privileged and require sudo to run services on

**Review by piiq — COMMENTED** [review]
This is pretty much fantastic. I've left a few minor comments here and there

File: openbb_platform_pro_backend/main.py (hunk 4 above)
Comment (deeleeramone) [reply openbb_platform_pro_backend/main.py]: Platform API is not reliant at all, however, every implementation I have encountered requires the key to be set as an environment variable and there are no uses where it is called from the Credentials model. So, for example, if `openbb-agents` is installed the key needs to be defined as such.

Tl;Dr: The key will not provide any benefits being defined in `user_settings.json` without also being added as an environment variable.

Comment (piiq) [reply openbb_platform_pro_backend/main.py]: yes, i understand that now. let's not have this as a part of the login script of the api. i suggest this is handled on the implementation level that requires it

File: openbb_platform_pro_backend/main.py (hunk 3 above)
Comment (deeleeramone) [reply openbb_platform_pro_backend/main.py]: Built-in methods allow for it not being visible at all.

**Review by piiq — APPROVED** [review]
